src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -689,6 +690,7 @@ internal async Task WriteDeltaNavigationPropertiesAsync(SelectExpandNode selectE
                         await WriteDynamicComplexPropertiesAsync(resourceContext, writer);
                         await WriteNavigationLinksAsync(selectExpandNode, resourceContext, writer);
                         await WriteExpandedNavigationPropertiesAsync(selectExpandNode, resourceContext, writer);
+                        await WriteNavigationPropertiesForDeepInsertAsync(selectExpandNode, resourceContext, writer);
```

Comment:
Currently we don't deserialize deltas in deep insert. If that changes, we will adjust accordingly.
We are calling this method inside `WriteResourceAsync` since nested resources in deep insert are deserialized as `T` not `Delta<T>`